tests/core/test_task_operators.py

```diff
@@ -85,6 +85,15 @@ def test_getitem_preserves_result_info(self):
         assert isinstance(y.result, LocalResult)
         assert y.result.dir.endswith("home")
 
+    def test_getitem_name(self):
+        with Flow(name="test") as f:
+            x = Parameter("x")[0]
+            assert x.name == "x[0]"
+            y = Parameter("y")["a"]
+            assert y.name == "y['a']"
+            z = Parameter("z")[Parameter("a")]
+            assert z.name == "z[<Parameter: a>]"
```

Comment:
Hmmm, this is a good callout, I think we could do better here. Perhaps?

```python
if isinstance(key, Task):
    name = f"{name}[{key.name}]"
else:
    name = f"{name}[{key!r}]"
```

Use the name of the task (as a non-repr) if the key is a task, otherwise repr the key as a constant.